Persephone/Components/Queue/QueueSongCoverView.swift

```diff
@@ -60,14 +60,15 @@ class QueueSongCoverView: NSTableCellView {
   }
 
   func setSong(_ queueItem: QueueItem, queueIcon: NSImage?) {
-    guard let imagePath = queueItem.song.album.coverArtFilePath
-      else { return }
+    let song = queueItem.song
 
     isPlaying = queueItem.isPlaying
-
-    let imageURL = URL(fileURLWithPath: imagePath)
-    let provider = LocalFileImageDataProvider(fileURL: imageURL)
     
+    let provider = MPDAlbumArtImageDataProvider(
+      songUri: song.mpdSong.uriString,
+      cacheKey: song.album.hash
+    )
```

Comment:
So it turns out I'm doing it this way because I don't always have access to the song itself, just the URI. The cacheKey does not need to be optional though, so I will change that.